weblab-identity-auth0/src/main/java/ucles/weblab/common/identity/auth0/UserAttributesResolver.java

```diff
@@ -34,18 +34,28 @@ public String getUserId() {
 
         if (authentication instanceof OAuth2Authentication) {
             OAuth2Authentication auth2Authentication = (OAuth2Authentication) authentication;
-            return auth2Authentication.getUserAuthentication().getName();
+            Object principal = auth2Authentication.getUserAuthentication().getPrincipal();
+
+            if (principal instanceof ExtendedUser) {
+                return getUserIdFromExtendedUser(principal);
+            } else {
+                return principal.toString();
+            }
         }
 
         if (authentication.getPrincipal() instanceof ExtendedUser) {
-            return (String) ((ExtendedUser) authentication.getPrincipal()).getMetadata().get("sub");
+            return getUserIdFromExtendedUser(authentication.getPrincipal());
         }
 
         String uid = "user|" + authentication.getName();
         log.warn("Returning fabricated id: {} for getUserId for Authentication: {}", uid, authentication.getClass().toString());
         return uid;
     }
 
+    private String getUserIdFromExtendedUser(Object principal) {
```

Comment:
Should be an ExtendedUser, not an Object.

Reply:
Done.